### PR TITLE
[DI-1307] - DRAFT - Adding DataModel refresh option

### PR DESCRIPTION
**Pull request by beto-devnet**
Adding a new button to refresh DataMaps and Resources tables

### CODE REVIEW

File: DataImport.Web/Features/ApiServers/RefreshDataModel.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                if (apiServer is null)
+                {
+                    throw new System.ArgumentException("Api server not found");
+                }
+
+                await _configurationService.FillSwaggerMetadata(apiServer);
+
+                await _database.SaveChangesAsync(cancellationToken);
+
+                var dataMaps = await (from dm in _database.DataMaps
```
Comment (stevenarnold-ed-fi) [inline DataImport.Web/Features/ApiServers/RefreshDataModel.cs:50]: @beto-devnet Please update to lambda syntax

File: DataImport.Web/Features/ApiServers/RefreshDataModel.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+                var metadataFromResources = await _database
+                    .Resources
+                    .Where(resource => resource.ApiVersionId == apiServer.ApiVersionId && resourcesPath.Contains(resource.Path))
+                    .Select(resource => new { resource.Path, resource.Metadata })
+                    .ToListAsync(cancellationToken);
+
+                dataMaps.ForEach(map =>
+                {
+                    map.Metadata = metadataFromResources.Where(x => x.Path == map.ResourcePath).Select(x => x.Metadata).First();
```
Comment (stevenarnold-ed-fi) [inline DataImport.Web/Features/ApiServers/RefreshDataModel.cs:66]: @beto-devnet .First() will throw an exception if there is more than one, is that appropriate here?

File: DataImport.Web/Features/ApiServers/RefreshDataModel.cs (hunk 1 above)
Comment (beto-devnet) [reply DataImport.Web/Features/ApiServers/RefreshDataModel.cs]: updated to labdasyntax

File: DataImport.Web/Features/ApiServers/RefreshDataModel.cs (hunk 2 above)
Comment (beto-devnet) [reply DataImport.Web/Features/ApiServers/RefreshDataModel.cs]: in this case, the issue could occur if the First() method returns an empty string, in that case, the Metadata should not be updated.

I modified the code and added a condition not to be updated if the string is empty

**Review by jasonh-edfi — APPROVED** [review]
Reviewed and approved as part of the community contribs.

**Pull request closed without merging**